run-build-functions.sh

```diff
@@ -655,13 +655,14 @@ install_dependencies() {
 
   if [ "$GIMME_GO_VERSION" != "$installGoVersion" ]
   then
-    echo "Installing Go version $installGoVersion"
-    GIMME_ENV_PREFIX=$HOME/.gimme_cache/env GIMME_VERSION_PREFIX=$HOME/.gimme_cache/versions gimme $installGoVersion
+    resolvedGoVersion=$(gimme --resolve $installGoVersion)
+    echo "Installing Go version $resolvedGoVersion (requested $installGoVersion)"
+    GIMME_ENV_PREFIX=$HOME/.gimme_cache/env GIMME_VERSION_PREFIX=$HOME/.gimme_cache/versions gimme $resolvedGoVersion
     if [ $? -eq 0 ]
     then
-      source $HOME/.gimme_cache/env/go$installGoVersion.linux.amd64.env
+      source $HOME/.gimme_cache/env/go$resolvedGoVersion.linux.amd64.env
     else
-      echo "Failed to install Go version '$installGoVersion'"
+      echo "Failed to install Go version '$resolvedGoVersion'"
       exit 1
     fi
   else
```

Comment:
Wondering if we could add a test case for this @akahn? (i.e. https://github.com/netlify/build-image/blob/focal/tests/node/base.bats#L13-L18). For more info around our tests see - https://github.com/netlify/build-image/blob/focal/CONTRIBUTING.md#tests